### ISSUE TITLE
Error 'Unsupported type 0x40 for array' when trying to complete deferred messages

### ISSUE THREAD

**Issue by lopezco** [Azure/azure-sdk-for-python#9010]
Hi everybody,

I'm having some trouble handling deferred messages. First I lost the sequence number which I recovered using *ServiceBusExplorer* as suggested [here][1]. Then, I used the same tool to resend the messages and they were correctly read. Nevertheless, the deferred copy of the messages are still in the queue and I cannot manage to delete them neither using *ServiceBusExplorer* nor using this client.

The error I'm getting is also mentioned in #8707 but the context is not the same.

I'm using Python 3.7 with `azure-servicebus==0.50.1`.
Here is the code example:

**File: logging.conf**
```
[loggers]
keys=root

[handlers]
keys=consoleHandler

[formatters]
keys=simpleFormatter

[logger_root]
level=INFO
handlers=consoleHandler

[handler_consoleHandler]
class=StreamHandler
level=INFO
formatter=simpleFormatter
args=(sys.stdout,)

[formatter_simpleFormatter]
format=[%(asctime)s] [%(levelname)s] [%(name)s]  %(message)s
datefmt=%Y-%m-%d %H:%M:%S
```

**main.py**
```python
import os
import logging.config
import logging

from azure.servicebus import QueueClient


logging.config.fileConfig(os.path.join('config', 'logging.conf'))
logger = logging.getLogger()


def handle_deferred_messages(connection_string, queue_name, sequence_numbers):
    client = QueueClient.from_connection_string(connection_string, queue_name)
    error_seq = []
    for s in sequence_numbers:
        with client.get_receiver() as queue_receiver:
            try:
                deferred_message = queue_receiver.receive_deferred_messages(sequence_numbers=[s])
            except Exception as e:
                logger.warning('Could not get message. Sequence number: {}'.format(s))
                continue
            else:
                logger.info('Message received. Sequence number: {}'.format(s))
                message = deferred_message[0]
                try:
                    message.complete()
                except Exception as e:
                    logger.error('Fail to complete message. Error: {}'.format(e))
                    try:
                        message.dead_letter()
                    except Exception as e:
                        logger.error('Fail to dead-letter message. Error: {}'.format(e))
                        error_seq.append(message)
                    else:
                        logger.error('Message sent to dead-letter. Sequence number: {}'.format(s))
                else:
                    logger.error('Message completed. Sequence number: {}'.format(s))
    return error_seq


if __name__ == '__main__':
    connection_string = '<Azure Service Bus connection>'
    queue_name = '<Azure Service Bus Queue Name>'
    errors = handle_deferred_messages(connection_string, queue_name, [3216, 3217])
```

When I execute `python main.py` with the correct `connection_string` (with *Manage*, *Send* and *Listen* rights) and the correct `queue_name`, I get the following output:

```
[2019-12-05 11:13:59] [INFO] [root]  Message received. Sequence number: 3216
[2019-12-05 11:13:59] [ERROR] [root]  Fail to complete message. Error: Management request failed: Management request returned status code: 500. Description: b'Unsupported type 0x40 for array. TrackingId:xxxxxxxx, SystemTracker:NoSystemTracker, Timestamp:2019-12-05T10:13:59', Data: None
[2019-12-05 11:13:59] [ERROR] [root]  Fail to dead-letter message. Error: Management request failed: Management request returned status code: 500. Description: b'Unsupported type 0x40 for array. TrackingId:xxxxxxxx, SystemTracker:NoSystemTracker, Timestamp:2019-12-05T10:13:59', Data: None
[2019-12-05 11:14:00] [INFO] [root]  Message received. Sequence number: 3217
[2019-12-05 11:14:00] [ERROR] [root]  Fail to complete message. Error: Management request failed: Management request returned status code: 500. Description: b'Unsupported type 0x40 for array. TrackingId:xxxxxxxx, SystemTracker:NoSystemTracker, Timestamp:2019-12-05T10:14:00', Data: None
[2019-12-05 11:14:00] [ERROR] [root]  Fail to dead-letter message. Error: Management request failed: Management request returned status code: 500. Description: b'Unsupported type 0x40 for array. TrackingId:xxxxxxxx, SystemTracker:NoSystemTracker, Timestamp:2019-12-05T10:14:00', Data: None
```

Traceback of the message.complete() call

```
Traceback (most recent call last):
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\base_handler.py", line 91, in _mgmt_request_response
    callback=callback)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\uamqp\client.py", line 358, in mgmt_request
    **kwargs)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\uamqp\session.py", line 126, in mgmt_request
    return parse_response(status, response, description)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\common\mgmt_handlers.py", line 18, in default
    status_code, description, message.get_data()))
azure.servicebus.common.errors.ServiceBusError: Management request returned status code: 500. Description: b'Unsupported type 0x40 for array. TrackingId:3afb23b6-09a3-425b-a7ea-fad9afc6cd44_G0, SystemTracker:NoSystemTracker, Timestamp:2019-12-05T11:34:26', Data: None
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\IPython\core\interactiveshell.py", line 3326, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-7a028d90601f>", line 53, in <module>
    errors = handle_deferred_messages(sequence_numbers)
  File "<ipython-input-2-7a028d90601f>", line 30, in handle_deferred_messages
    raise e
  File "<ipython-input-2-7a028d90601f>", line 27, in handle_deferred_messages
    message.complete()
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\common\message.py", line 492, in complete
    self._receiver._settle_deferred('completed', [self.lock_token])  # pylint: disable=protected-access
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\receive_handler.py", line 152, in _settle_deferred
    mgmt_handlers.default)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\base_handler.py", line 93, in _mgmt_request_response
    raise ServiceBusError("Management request failed: {}".format(exp), exp)
azure.servicebus.common.errors.ServiceBusError: Management request failed: Management request returned status code: 500. Description: b'Unsupported type 0x40 for array. TrackingId:3afb23b6-09a3-425b-a7ea-fad9afc6cd44_G0, SystemTracker:NoSystemTracker, Timestamp:2019-12-05T11:34:26', Data: None
```

Traceback of the message.dead_letter() call

```
Traceback (most recent call last):
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\base_handler.py", line 91, in _mgmt_request_response
    callback=callback)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\uamqp\client.py", line 358, in mgmt_request
    **kwargs)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\uamqp\session.py", line 126, in mgmt_request
    return parse_response(status, response, description)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\common\mgmt_handlers.py", line 18, in default
    status_code, description, message.get_data()))
azure.servicebus.common.errors.ServiceBusError: Management request returned status code: 500. Description: b'Unsupported type 0x40 for array. TrackingId:47ce260f-b464-4c09-9c2a-fb1ff20ce91f_G5, SystemTracker:NoSystemTracker, Timestamp:2019-12-05T12:57:23', Data: None
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<ipython-input-2-4e7f4ed4ea1d>", line 27, in handle_deferred_messages
    message.complete()
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\common\message.py", line 492, in complete
    self._receiver._settle_deferred('completed', [self.lock_token])  # pylint: disable=protected-access
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\receive_handler.py", line 152, in _settle_deferred
    mgmt_handlers.default)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\base_handler.py", line 93, in _mgmt_request_response
    raise ServiceBusError("Management request failed: {}".format(exp), exp)
azure.servicebus.common.errors.ServiceBusError: Management request failed: Management request returned status code: 500. Description: b'Unsupported type 0x40 for array. TrackingId:47ce260f-b464-4c09-9c2a-fb1ff20ce91f_G5, SystemTracker:NoSystemTracker, Timestamp:2019-12-05T12:57:23', Data: None
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\base_handler.py", line 91, in _mgmt_request_response
    callback=callback)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\uamqp\client.py", line 358, in mgmt_request
    **kwargs)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\uamqp\session.py", line 126, in mgmt_request
    return parse_response(status, response, description)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\common\mgmt_handlers.py", line 18, in default
    status_code, description, message.get_data()))
azure.servicebus.common.errors.ServiceBusError: Management request returned status code: 500. Description: b'Unsupported type 0x40 for array. TrackingId:fa259905-24e2-4a7a-8ad5-a8b237c092f5_G5, SystemTracker:NoSystemTracker, Timestamp:2019-12-05T12:57:23', Data: None
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\IPython\core\interactiveshell.py", line 3326, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-4e7f4ed4ea1d>", line 53, in <module>
    errors = handle_deferred_messages(sequence_numbers)
  File "<ipython-input-2-4e7f4ed4ea1d>", line 35, in handle_deferred_messages
    raise e
  File "<ipython-input-2-4e7f4ed4ea1d>", line 31, in handle_deferred_messages
    message.dead_letter()
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\common\message.py", line 516, in dead_letter
    'suspended', [self.lock_token], dead_letter_details=details)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\receive_handler.py", line 152, in _settle_deferred
    mgmt_handlers.default)
  File "C:\Users\my_user\.conda\envs\event-consumer\lib\site-packages\azure\servicebus\base_handler.py", line 93, in _mgmt_request_response
    raise ServiceBusError("Management request failed: {}".format(exp), exp)
azure.servicebus.common.errors.ServiceBusError: Management request failed: Management request returned status code: 500. Description: b'Unsupported type 0x40 for array. TrackingId:fa259905-24e2-4a7a-8ad5-a8b237c092f5_G5, SystemTracker:NoSystemTracker, Timestamp:2019-12-05T12:57:23', Data: None
```
Do you have any suggestions?

Thanks.

  [1]: https://stackoverflow.com/a/46070371/12484641

**Comment by KieranBrantnerMagee** [comment]
Hey lopezco, thanks for reporting this.

Given the issue chain you referenced and similarities in the failure mode, I'm very interested to see if this problem would be mitigated by the release we will be shipping shortly.

We're currently in the final stages of validation and preparation, and intend to releasing within the next week.  If that doesn't mitigate the issue, we'll certainly dig in further.

Don't hesitate to let us know if there is urgency here and I can see about providing a preview whl for you to evaluate sooner.

**Comment by lopezco** [comment]
Hi @KieranBrantnerMagee , thanks for your answer.

It is not critical because I've managed to copy the messages so no data was lost.

I also cleaned the deferred messages with a `time.sleep()` to force a longer processing time than the lock time. After this, they were automatically transferred to the dead-letter queue where I was able to delete them using the *ServiceBusExplorer*.

**Comment by KieranBrantnerMagee** [comment]
Looping back on this;  we released a new SB release (0.50.2) prior to the holidays per my earlier mention, am wondering if this mitigated the issues you observed.

Don't hesitate to ping if not/if any clarification or help is needed and we'll gladly dig in; thanks as always for your time!
